### PR TITLE
Changed example showing gradient processing

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
The original example was processing the gradients twice.
1st: grads_and_vars = zip(processed_grads, var_list)
2nd: capped_grads_and_vars = [(MyCapper(gv[0]), gv[1]) for gv in grads_and_vars]

The 2nd line is especially weird because it unnecessarily zips the gradients with the var_list even though it is only processing the gradient part.

Refactored the example to be clearer, now there is only a single line that processes the gradients.

### CODE REVIEW

**Comment by HarikrishnanBalagopal** [comment]
@googlebot I signed it!

File: tensorflow/python/keras/optimizer_v2/optimizer_v2.py [hunk 1]
```diff
@@ -138,15 +138,13 @@ class directly, but instead instantiate one of its subclasses such as
     loss = <call_loss_function>
   vars = <list_of_variables>
   grads = tape.gradient(loss, vars)
-  processed_grads = [process_gradient(g) for g in grads]
-  grads_and_vars = zip(processed_grads, var_list)
 
-  # grads_and_vars is a list of tuples (gradient, variable).  Do whatever you
-  # need to the 'gradient' part, for example cap them, etc.
-  capped_grads_and_vars = [(MyCapper(gv[0]), gv[1]) for gv in grads_and_vars]
+  # Do whatever you need to the gradients, for example cap them, etc.
```
Comment (KabirKwatra) [inline tensorflow/python/keras/optimizer_v2/optimizer_v2.py]: `Apply pre-processing` or `Process` instead of `Do whatever you need`

Comment (HarikrishnanBalagopal) [reply tensorflow/python/keras/optimizer_v2/optimizer_v2.py]: Sure. How do I change it?

**Review by tanzhenyu — APPROVED** [review]
